xlators/features/index/src/index.c

```diff
@@ -269,19 +266,17 @@ index_dir_create(xlator_t *this, const char *subdir)
         goto out;
     }
 
-    pathlen = strlen(fullpath);
-    if ((pathlen > 1) && fullpath[pathlen - 1] == '/')
-        fullpath[pathlen - 1] = '\0';
+    gf_path_strip_trailing_slashes(fullpath);
```

Comment:
Looks like in all cases we might be able to skip the strlen() since the string length is already known. Specifically for the index xlator, it's worthwhile (perhaps in another patch!) to have make_index_dir_path() return the string length (which is the return value of that snprintf() anyway).